src/task/task_graph.rs

```diff
@@ -118,6 +118,7 @@ impl<'p> TaskGraph<'p> {
         project: &'p Project,
         search_envs: &SearchEnvironments<'p, D>,
         args: Vec<String>,
+        isolated: bool,
```

Comment:
I think in terms of responsibility its a bit strange that the task graph needs to be aware of isolation. Cant we default it to `false` for the custom task?